marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/writer/HubDataWriter.java

```diff
@@ -61,7 +62,15 @@ public HubDataWriter(HubClient hubClient, long taskId, StructType schema, Map<St
             hubClient.getModulesClient().newTextDocumentManager().read(apiModulePath, new StringHandle())
         ).bulkCaller();
 
-        loader.setWorkUnit(new ByteArrayInputStream(("{\"taskId\":" + taskId + "}").getBytes()));
+        String workUnit = "{\"taskId\":" + taskId + "}";
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            workUnit = workUnit.concat(objectMapper.writeValueAsString(params));
```

Comment:
I don't think it's safe to write all of the params as a JSON object, as that could contain the user password and other sensitive information. 

We should be much more precise about what goes into workUnit. So we should have a method of e.g. "ObjectNode buildWorkUnitFromParams(Map)" that builds an ObjectNode based on the params we know may exist in the Map, and nothing more than that.